jekyll/_cci2/creating-orbs.md

```diff
@@ -0,0 +1,332 @@
+
+A development version must be referenced entirely, such as: ```mynamespace/myorb@dev:mybranch.```; whereas production orbs allow wildcard semantic version references. Note that there are no shorthand conveniences for development versions.
+
+<aside class="notice">
+Dev versions are mutable and expire: their contents can change, and they are subject to deletion after 90 days; therefore, it is strongly recommended you do not rely on a development versions in any production software, and use them only while actively developing your orb.
+</aside>
+
+## Importing an Orb
+
+Importing a set of existing orbs is a simple process that only requires you to provide the following instruction in your configuration. In the example below, you can see how to import a CircleCI Slack and Heroku orb:
+
+```orbs:
+  slack: circleci/slack@0.1.0
+  heroku: circleci/heroku@volatile
+```
+
+### Designing Orbs
+
+When designing your own orbs, make sure your orbs meet the following requirements:
+
+* Orbs should always use 'description'. - Be sure to explain usage, assumptions, and any tricks in the ```description``` key under jobs, commands, executors, and parameters.
+* Match commands to executors - If you are providing commands, try to provide one or more executors in which they will run.
+* Use concise naming for your Orb - Remember that use of your commands and jobs is always contextual to your Orb, so you can use general names like "run-tests" in most cases.
+* Required vs. Optional Parameters - Provide sound default values of parameters whenever possible.
+* Avoid Job-only Orbs - Job-only Orbs are inflexible. While these Orbs are sometimes appropriate, it can be frustrating for users to not be able to use the commands in their own jobs. Pre and post steps when invoking jobs are a workaround for users.
+* Parameter `steps` are powerful - Wrapping steps provided by the user allows you to encapsulate and sugar things like caching strategies and other more complex tasks, providing a lot of value to users.
+
+When developing your own orb, you may find it useful to write an inline orb. The section below describes how you can write your own inline orb.
+
+## Creating Inline Orbs
+
+Inline orbs can be handy during development of an orb or as a convenience for name-spacing jobs and commands in lengthy configurations, particularly if you later intend to share the orb with others.
+
+To write inline orbs, you need to place the orb elements under that orb's key in the ```orbs``` declaration in the configuration. For example, if you want to import one orb and then author inline for another, the orb might look like the example shown below:
+
+```
+orbs:
+  codecov: circleci/codecov-clojure@0.0.4
+  my-orb:
+    executors:
+      default:
+        docker:
+          - image: circleci/ruby:1.4.2
+    commands:
+      dospecialthings:
+        steps:
+          - run: echo "We will now do special things"
+    jobs:
+      myjob:
+        executor: default
```

Comment:
Could we call this executor something other than `default`? `default` feels like one of those words that might be required, vs. one that's clearly and example word.

Reply:
changed